### PR TITLE
Optimize code that consume most resources

### PR DESCRIPTION
**Pull request by markus-gsf-seidl**
Warning: This can also be seen as a meta-pattern. Why? Because "Optimize the application" is in itself a very generic term.

#59

### CODE REVIEW

**Comment by dubrie** [comment]
Looks good for initial review.

**Comment by dubrie** [comment]
Assigning to @greenhsu123 for SME review

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, optimize application components, will impact two parts:
+
+- `E`: If memory, amount of used CPUs or storage can be decreased, then the embodied emissions will be decreased
+- `I`: If CPU usage can be decreased the software intensity will be decreased
```
Comment (greenhsu123) [inline docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:32]: I would argue I is carbon intensity and not software intensity. Thoughts?

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Assumptions
+
+- Optimization needs proficiency in the tools and time to be implemented.
+
+## Considerations
+
+- Consider the other cloud patterns in this category
+
+## References
```
Comment (greenhsu123) [inline docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:42]: I think we should also add energy proportionality here as a reference. Thoughts?

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- `E`: If memory, amount of used CPUs or storage can be decreased, then the embodied emissions will be decreased
+- `I`: If CPU usage can be decreased the software intensity will be decreased
+
+## Assumptions
+
+- Optimization needs proficiency in the tools and time to be implemented.
+
+## Considerations
+
+- Consider the other cloud patterns in this category
```
Comment (greenhsu123) [inline docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:40]: This consideration doesn't make much sense, IMO, as we have many patterns in the cloud category. Could you explain further?

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ - size:medium
+---
+
+# Optimize application components that consume the most resources
+
+## Description
+
+Monitor the application to identify components that consume the most time or resources. Optimize the code or architecture so, that these component minimize the resource usage while maximizing performance.
+
+## Solution
```
[the remark this answers is not in the document]
Comment (markus-ntt-seidl) [reply docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:18]: I don't see a splitting path, can you elaborate?

PS: Hm, I was looking at your other reviews and I have the feeling that you suggest this if there are two sentences in the solution. Are you against two sentences in the solution?

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md (hunk 1 above)
Comment (franziska-warncke) [reply docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:32]: agree!

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md (hunk 2 above)
Comment (franziska-warncke) [reply docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:42]: Can you explain here a little bit more, what you want here?

File: docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (franziska-warncke) [reply docs/catalog/cloud/optimize-application-components-that-consome-the-most-resources.md:18]: I do not see a splitting path here too. But maybe you can give us more detail here :)

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl Can you please have a look again

**Pull request closed without merging**